examples/plot_compare_reduction.py

```diff
@@ -72,4 +86,45 @@
 plt.ylabel('Digit classification accuracy')
 plt.ylim((0, 1))
 plt.legend(loc='upper left')
+
+###############################################################################
+# Caching transformers within a ``Pipeline``
```

Comment:
I'm not very happy with this appended to the existing example. Not sure what the right solution is, though.